### ISSUE TITLE
Cached QueryStream with multiple listeners emits stale data

### ISSUE THREAD

**Issue by Mike278** [simolus3/moor#166]
```dart
    // in .moor file: create table tbl (col1 text, col2 int, col3 int);

    Stream<List<TblData>> all = db.select(db.tbl).watch();
    all.listen(print); // Comment out this line to make test pass

    await db.into(db.tbl).insert(TblData(col1: 'test'));
    List<TblData> data = await db.select(db.tbl).watch().first;
    expect(data.length, 1);
    expect(data.first.col1, 'test');
```

**Comment by simolus3** [comment]
Thanks for the report. The situation should be better on `develop` now, we delete cached data from the stream when the table updates. It's a bit tricky to get right, as the future returned by the `insert` statement could complete before the streams were updated.

I made the `QueryStream` always skip a microtask before checking for cached data and emitting it. (Previously we'd check for cached data right away and then emit it a microtask later).
